packages/react-core/src/components/Nav/NavItem.tsx

```diff
@@ -39,12 +54,148 @@ export const NavItem: React.FunctionComponent<NavItemProps> = ({
   preventDefault = false,
   onClick = null as NavSelectClickHandler,
   component = 'a',
+  flyout,
+  onShowFlyout,
   ouiaId,
   ouiaSafe,
   ...props
 }: NavItemProps) => {
-  const Component = component as any;
+  const { flyoutRef, setFlyoutRef, disableHover } = React.useContext(NavContext);
   const { isNavOpen } = React.useContext(PageSidebarContext);
+  const flyoutContext = React.useContext(FlyoutContext);
+  const [flyoutXDirection, setFlyoutXDirection] = React.useState(flyoutContext.direction);
+  const ref = React.useRef<HTMLLIElement>();
+  const flyoutVisible = ref === flyoutRef;
+  const Component = component as any;
+  const hasFlyout = flyout !== undefined;
+
+  const showFlyout = (show: boolean) => {
+    if (!flyoutVisible && show) {
+      setFlyoutRef(ref);
+    } else if (flyoutVisible && !show) {
+      setFlyoutRef(null);
+    }
+    onShowFlyout && show && onShowFlyout();
+  };
+
+  useIsomorphicLayoutEffect(() => {
+    if (hasFlyout && ref.current && canUseDOM) {
```

Comment:
how similar is this to the menu?  is there any way to share some come functionality in s utility?  If so we can follow up on that.

Reply:
This is the same function from Menu (determining when to flip the flyout direction), but it doesn't work quite right for Nav at the moment. The styles it applies to flip the direction left doesn't work with the Nav item. We could make this a utility, or I could remove it for now.

Reply:
+1 for keeping flyout code DRY. It was very difficult to write the first time around and I think it hard enough to maintain a single copy.

Maybe a set of our own `useFlyout` hooks would work?

Reply:
The classes applied would be different (menu styles for Menu, currently nothing for Nav) but I can abstract the logic of the function into a util

Reply:
Or can you pass properties to hooks directly?

Reply:
You can pass whatever you want to custom hooks! See https://reactjs.org/docs/hooks-custom.html

Reply:
I'm half inclined to remove the whole useEffect for Nav though, because it doesn't actually work currently. Opinions?

Reply:
Less code is better!